main/src/cgeo/geocaching/settings/OCMaxDistancePreference.java

```diff
@@ -0,0 +1,112 @@
+package cgeo.geocaching.settings;
+
+import cgeo.geocaching.R;
+import cgeo.geocaching.location.IConversion;
+
+import android.content.Context;
+import android.preference.Preference;
+import android.util.AttributeSet;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.SeekBar;
+import android.widget.SeekBar.OnSeekBarChangeListener;
+import android.widget.TextView;
+
+import java.util.Locale;
+
+import butterknife.ButterKnife;
+
+public class OCMaxDistancePreference extends Preference {
+
+    private int seekbarLength;
+    private TextView valueView;
+
+    public OCMaxDistancePreference(final Context context) {
+        super(context);
+        init();
+    }
+
+    public OCMaxDistancePreference(final Context context, final AttributeSet attrs) {
+        super(context, attrs);
+        init();
+    }
+
+    public OCMaxDistancePreference(final Context context, final AttributeSet attrs, final int defStyle) {
+        super(context, attrs, defStyle);
+        init();
+    }
+
+    private void init() {
+        setPersistent(false);
+    }
+
+    private int km2valueHelper(final double distance) {
```

Comment:
Doesn't seem to be belonging to `settings`. There was some `util`-like package, please put it there

Reply:
This conversion formula is specific to the seekbar length and the way that specific seekbar accelerates. (Android only offers linear seekbars, but by using those two formulas I made it exponential, which suits better for distance setting over such a large range, IMHO.) That's why I put it into this specific seekbar class.